cns/restserver/restserver.go

```diff
@@ -211,7 +211,17 @@ func NewHTTPRestService(config *common.ServiceConfig, wscli interfaceGetter, wsp
 
 // Init starts the CNS listener.
 func (service *HTTPRestService) Init(config *common.ServiceConfig) error {
-	err := service.Initialize(config)
+	res, err := service.wscli.GetInterfaces(context.TODO())
+	if err != nil {
+		return errors.Wrap(err, "failed to get interfaces from IMDS")
+	}
+	primaryInterface, err := wireserver.GetPrimaryInterfaceFromResult(res)
+	if err != nil {
+		return errors.Wrap(err, "failed to get primary interface from IMDS response")
+	}
+
+	err = service.Initialize(config, primaryInterface.PrimaryIP)
```

Comment:
i think we should put the primary interface's primary IP inside the config. The `primaryInterface.PrimaryIP` seems to be a detail that the HTTPRestService could potentially derive from the config, rather than requiring it to be passed in explicitly.